src/pages/tokens/[id].tsx

```diff
@@ -0,0 +1,203 @@
+import { PoolPairsTable } from '@components/commons/tables/PoolPairs'
+
+interface TokenAssetPageProps {
+  token: TokenData
+}
+
+export default function TokenIdPage (props: InferGetServerSidePropsType<typeof getServerSideProps>): JSX.Element {
+  // TODO(@siradji) Fetch poolpairs in upcoming upstreaming
+  const poolpairs: any = [
+    {
+      id: '4',
+      symbol: 'ETH-DFI',
+      name: 'Ether-Default Defi token',
+      status: true,
+      tokenA: {
+        symbol: 'ETH',
+        id: '1',
+        reserve: '9780.54512445',
+        blockCommission: '0'
+      },
+      tokenB: {
+        symbol: 'DFI',
+        id: '0',
+        reserve: '11036692.66519257',
+        blockCommission: '0'
+      },
+      priceRatio: { ab: '0.00088618', ba: '1128.43328513' },
+      commission: '0.002',
+      totalLiquidity: { token: '328409.39535062', usd: '62057496.7252094383473678' },
+      tradeEnabled: true,
+      ownerAddress: '8UAhRuUFCyFUHEPD7qvtj8Zy2HxF5HH5nb',
+      rewardPct: '0.14549',
+      creation: {
+        tx: '9827894c083b77938d13884f0404539daa054a818e0c5019afa1eeff0437a51b',
+        height: 466822
+      },
+      apr: { reward: 0.6353433769288943, total: 0.6353433769288943 }
+    }
+  ]
+
+  return (
+    <div className='container mx-auto px-4 pt-12 pb-20'>
+      <Breadcrumb items={[
+        { path: '/tokens', name: 'Tokens' },
+        { path: `/tokens/${props.token.id}`, name: `${props.token.name}`, hide: false, canonical: true }
+      ]}
+      />
+      <PageHeading token={props.token} />
+      <div className='flex flex-col md:flex-row md:space-x-8'>
+        <ListLeft token={props.token} />
+        <ListRight token={props.token} />
+      </div>
+      <div className='mt-10'>
+        <h1 className='text-2xl font-semibold'>Token Pairs</h1>
+        <PoolPairsTable poolPairs={poolpairs} />
+      </div>
+    </div>
+  )
+}
+
+function PageHeading ({ token }: { token: TokenData }): JSX.Element {
+  if (token.isDAT) {
+    const AssetIcon = getAssetIcon(token.symbol)
+    return (
+      <div className='flex flex-row flex-wrap items-center my-4'>
+        <AssetIcon className='h-8 w-8 mr-4' />
+        <h1 data-testid='PageHeading' className='text-2xl font-semibold'>{token.name.replace('Default Defi token', 'DeFiChain')}</h1>
+      </div>
+    )
+  }
+  const TokenIcon = getTokenIcon(token.symbol)
+  return (
+    <div className='flex flex-row flex-wrap items-center my-4'>
+      <TokenIcon className='h-8 w-8 mr-2' />
+      <h1 className='text-2xl font-semibold'>{token.name}</h1>
+    </div>
+  )
+}
+
+function ListRight ({ token }: { token: TokenData }): JSX.Element {
+  return (
+    <AdaptiveList>
+      <AdaptiveList.Row>
+        <AdaptiveList.Cell name='Decimal'>{token.decimal} Places</AdaptiveList.Cell>
+      </AdaptiveList.Row>
+      <AdaptiveList.Row>
+        <AdaptiveList.Cell name='Limit'>{token.limit}</AdaptiveList.Cell>
+      </AdaptiveList.Row>
```

Comment:
Linguistically from the design, there is only `<Row/>`s in this AdaptiveList. This greatly reduces your HTML node horizontal indentation staircase `<><><>`. I would opt against preemptively making it too abstract, you should design specific atomic component instead of trying to achieve too many things.

it should be:

```html
<AdaptiveList.Row name='Decimal'>{token.decimal} Places</AdaptiveList.Cell>
<AdaptiveList.Row name='Limit'>{token.limit}</AdaptiveList.Cell>
```